relayer/src/foreign_client.rs

```diff
@@ -432,34 +463,6 @@ impl ForeignClient {
                 ))
             })?;
 
-        // If not specified, set trusted state to the highest height smaller than target height.
-        // Otherwise ensure that a consensus state at trusted height exists on-chain.
-        let cs_heights = self.consensus_state_heights()?;
-        let trusted_height = if trusted_height == Height::zero() {
```

Comment:
Since we don't have this check in place anymore, this means that any caller of `build_update_client_with_trusted` has to make sure that the client has a consensus state at `trusted_height` installed. Do I understand correctly?

Reply:
Correct. In pretty much all places, the trusted height is already obtained from the consensus state height, so the check is unnecessary. With the new `update client` command, I make it such that the trusted height is also automatically fetchd, so there is no need for checking. The only place left unchecked is in the `tx raw update-client` command, but since it is a raw command I think it is good to bypass checking so that we can test the behavior on the actual chain.